### PR TITLE
[Auto Parallel] Support pure single-card modeling for auto_parallel

### PR DESCRIPTION
**Pull request by haohongxiang**
### PR types

Function optimization
### PR changes

Others
### Description

[Auto Parallel] Support pure single-card modeling for auto_parallel

### CODE REVIEW

File: paddlenlp/transformers/llama/modeling_auto.py [hunk 1]
```diff
@@ -0,0 +1,1238 @@
...
+        if attention_mask.shape != [bsz, 1, q_len, kv_seq_len]:
+            raise ValueError(
+                f"Attention mask should be of shape {(bsz, 1, q_len, kv_seq_len)}, but is {attention_mask.shape}"
+            )
+
+        attn_weights = attn_weights + attention_mask
+        if not paddle.in_dynamic_mode():
+            attn_weights = F.softmax(attn_weights, axis=-1, dtype="float32").astype(query_states.dtype)
+        else:
+            with paddle.amp.auto_cast(False):
```
Comment (zhaoyinglia) [inline paddlenlp/transformers/llama/modeling_auto.py]: 有auto_cast的地方目前也需要删。动转静会挂掉

Comment (haohongxiang) [reply paddlenlp/transformers/llama/modeling_auto.py]: Done

**Review by zhaoyinglia — APPROVED** [review]
LTGM

**Review by zhaoyinglia — APPROVED** [review]

**Review by wawltor — APPROVED** [review]
LGTM

**Pull request merged**